src/o365/flow/commands/flow-remove.ts

```diff
@@ -0,0 +1,171 @@
+import commands from '../commands';
+import GlobalOptions from '../../../GlobalOptions';
+import {
+  CommandOption,
+  CommandValidate
+} from '../../../Command';
+import request from '../../../request';
+import AzmgmtCommand from '../../base/AzmgmtCommand';
+import * as os from 'os';
+import Utils from '../../../Utils';
+
+const vorpal: Vorpal = require('../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  environment: string;
+  name: string;
+  asAdmin?: boolean;
+  confirm?: boolean;
+}
+
+class FlowRemoveCommand extends AzmgmtCommand {
+  public get name(): string {
+    return commands.FLOW_REMOVE;
+  }
+
+  public get description(): string {
+    return 'Removes the specified Microsoft Flow';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.asAdmin = typeof args.options.asAdmin !== 'undefined';
+    telemetryProps.confirm = typeof args.options.confirm !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    if (this.verbose) {
+      cmd.log(`Removing Microsoft Flow ${args.options.name}...`);
+    }
+    const removeFlow: () => void = (): void => {
+      const requestOptions: any = {
+        url: `${this.resource}providers/Microsoft.ProcessSimple/${args.options.asAdmin ? 'scopes/admin/' : ''}environments/${encodeURIComponent(args.options.environment)}/flows/${encodeURIComponent(args.options.name)}?api-version=2016-11-01`,
+        resolveWithFullResponse: true,
+        headers: {
+          accept: 'application/json'
+        },
+        json: true
+      };
+
+      request
+        .delete(requestOptions)
+        .then((rawRes: any): void => {
+          // api bug, should return 404 but instead returns 204
```

Comment:
You might want to change this comment based on the feedback from Microsoft.

Reply:
Comment updated and added URL to comment in issue that contains response from Microsoft support